### PR TITLE
Remove usage of github app token in structurizr-render-diagrams

### PR DESCRIPTION
**Pull request by Renetnielsen**

### CODE REVIEW

File: .github/workflows/structurizr-render-diagrams.yml [hunk 1]
```diff
@@ -73,20 +77,36 @@ jobs:
...
+          private_key: ${{ secrets.dh3serviceaccount_privatekey }
+
+      - name: Checkout
+        if: ${{ secrets.dh3serviceaccount_privatekey != '' }}
+        uses: actions/checkout@v4
+        with:
+          # Repository checkout must be called with a PAT token with `repo` and `workflow` scopes.
+          # Otherwise `stefanzweifel/git-auto-commit-action` used later in the workflow won't trigger new workflow runs.
+          # Link: https://github.com/stefanzweifel/git-auto-commit-action#commits-made-by-this-action-do-not-trigger-new-workflow-runs
+          token: ${{ secrets.token }}
```
Comment (djorgensendk) [inline .github/workflows/structurizr-render-diagrams.yml:97]:
```suggestion
          token: ${{ steps.generate_token.outputs.token }}
```

File: .github/workflows/structurizr-render-diagrams.yml [hunk 2]
```diff
@@ -62,7 +62,11 @@ on:
       dh3serviceaccount_privatekey:
         description: |
           Used to re-trigger workflows if rendered diagrams are pushed.
-        required: true
+        required: false
+      token:
+        description: |
+          Used to re-trigger workflows if rendered diagrams are pushed.
+        required: false
```
Comment (djorgensendk) [inline .github/workflows/structurizr-render-diagrams.yml:69]: descriptions doesn't make alot of sense

File: .github/workflows/structurizr-render-diagrams.yml [hunk 3]
```diff
@@ -62,7 +62,11 @@ on:
       dh3serviceaccount_privatekey:
         description: |
           Used to re-trigger workflows if rendered diagrams are pushed.
-        required: true
+        required: false
+      token:
+        description: |
```
Comment (Renetnielsen) [inline .github/workflows/structurizr-render-diagrams.yml:67]:
```suggestion
        description: Token used to checkout the repository
```

File: .github/workflows/structurizr-render-diagrams.yml [hunk 4]
```diff
@@ -62,7 +62,11 @@ on:
       dh3serviceaccount_privatekey:
         description: |
           Used to re-trigger workflows if rendered diagrams are pushed.
-        required: true
+        required: false
+      token:
+        description: Token used to checkout the repository
+          Used to re-trigger workflows if rendered diagrams are pushed.
```
Comment (djorgensendk) [inline .github/workflows/structurizr-render-diagrams.yml:68]:
```suggestion
      token:
        description: Token used to checkout the repository
```

**Review by djorgensendk — APPROVED** [review]

**Pull request merged**